### ISSUE TITLE
Default DNS search domains not being configured after systemd v242 update

### ISSUE THREAD

**Issue by puneetse** [clearlinux/distribution#923]
systemd-resolved does not seem to be populating /etc/resolve.conf with the default search domain since the merge of systemd-242 last week.
 
This causes local DNS names to not be resolved, like pacdiscovery finding wpad.dat information.

**Comment by puneetse** [comment]
Working in version 29880 (systemd v241)
Not working in 29910 (systemd v242)

**Comment by ahkok** [comment]
How about the releases in between?

**Comment by puneetse** [comment]
Not working in 29890 either.
I'm running with SYSTEMD_LOG_LEVEL=debug to see if there is any info there.

**Comment by puneetse** [comment]
Logs from 29890 (not working):
```
$ sudo SYSTEMD_LOG_LEVEL=debug /lib/systemd/systemd-resolved
Positive Trust Anchors:
...
Negative trust anchors: ...
Failed to open configuration file '/etc/systemd/resolved.conf': No such file or directory
Using system hostname 'clr-...'.
New scope on link *, protocol dns, family *
Found new link 3/wlp6s0
Found new link 2/enp11s0
Found new link 1/lo
New scope on link enp11s0, protocol dns, family *
Bus bus-api-resolve: changing state UNSET → OPENING
Bus bus-api-resolve: changing state OPENING → AUTHENTICATING
Creating stub listener using UDP/TCP.
Bus bus-api-resolve: changing state AUTHENTICATING → HELLO
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello cookie=1 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RequestName cookie=2 reply_cookie=0 signature=su error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=3 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.96 path=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 signature=s error-name=n/a error-message=n/a
Bus bus-api-resolve: changing state HELLO → RUNNING
Got message type=signal sender=org.freedesktop.DBus.Local destination=n/a path=/org/freedesktop/DBus/Local interface=org.freedesktop.DBus.Local member=Connected cookie=4294967295 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Got message type=signal sender=org.freedesktop.DBus destination=:1.96 path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=2 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=signal sender=org.freedesktop.DBus destination=:1.96 path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=3 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.96 path=n/a interface=n/a member=n/a cookie=4 reply_cookie=2 signature=u error-name=n/a error-message=n/a
Successfully acquired requested service name.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.96 path=n/a interface=n/a member=n/a cookie=5 reply_cookie=3 signature=n/a error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.login1',path='/org/freedesktop/login1',interface='org.freedesktop.login1.Manager',member='PrepareForSleep' successfully installed.
Got message type=method_call sender=:1.97 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=ResolveHostname cookie=2 reply_cookie=0 signature=isit error-name=n/a error-message=n/a
Looking up RR for 2.clearlinux.pool.ntp.org IN A.
Looking up RR for 2.clearlinux.pool.ntp.org IN AAAA.
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=4 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetNameOwner cookie=5 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.96 path=n/a interface=n/a member=n/a cookie=7 reply_cookie=5 signature=s error-name=n/a error-message=n/a
Switching to DNS server 10.x.x.x for interface enp11s0.
Cache miss for 2.clearlinux.pool.ntp.org IN A
Transaction 9645 for <2.clearlinux.pool.ntp.org IN A> scope dns on enp11s0/*.
Using feature level UDP+EDNS0 for transaction 9645.
Using DNS server 10.x.x.x for transaction 9645.
Sending query packet with id 9645.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.96 path=n/a interface=n/a member=n/a cookie=6 reply_cookie=4 signature=n/a error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.DBus',path='/org/freedesktop/DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0=':1.97' successfully installed.
Processing incoming packet on transaction 9645 (rcode=SUCCESS).
Verified we get a response at feature level UDP+EDNS0 from DNS server 10.x.x.x.
Added positive unauthenticated cache entry for 2.clearlinux.pool.ntp.org IN A 132s on enp11s0/INET/10.x.x.x
Added positive unauthenticated cache entry for 2.clearlinux.pool.ntp.org IN A 132s on enp11s0/INET/10.x.x.x
Added positive unauthenticated cache entry for 2.clearlinux.pool.ntp.org IN A 132s on enp11s0/INET/10.x.x.x
Added positive unauthenticated cache entry for 2.clearlinux.pool.ntp.org IN A 132s on enp11s0/INET/10.x.x.x
Transaction 9645 for <2.clearlinux.pool.ntp.org IN A> on scope dns on enp11s0/* now complete with <success> from network (unsigned).
Sent message type=method_return sender=n/a destination=:1.97 path=n/a interface=n/a member=n/a cookie=6 reply_cookie=2 signature=a(iiay)st error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=7 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Freeing transaction 9645.
```

Logs from 29880 (working):
```

 $ sudo SYSTEMD_LOG_LEVEL=debug /lib/systemd/systemd-resolved
Positive Trust Anchors:
...
Negative trust anchors: ...
Failed to open configuration file '/etc/systemd/resolved.conf': No such file or directory
Using system hostname 'clr-...'.
New scope on link *, protocol dns, family *
Found new link 3/wlp6s0
Found new link 2/enp11s0
Found new link 1/lo
New scope on link enp11s0, protocol dns, family *
Bus bus-api-resolve: changing state UNSET → OPENING
Bus bus-api-resolve: changing state OPENING → AUTHENTICATING
Creating stub listener using UDP/TCP.
Bus bus-api-resolve: changing state AUTHENTICATING → HELLO
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello cookie=1 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RequestName cookie=2 reply_cookie=0 signature=su error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=3 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 signature=s error-name=n/a error-message=n/a
Bus bus-api-resolve: changing state HELLO → RUNNING
Got message type=signal sender=org.freedesktop.DBus.Local destination=n/a path=/org/freedesktop/DBus/Local interface=org.freedesktop.DBus.Local member=Connected cookie=4294967295 reply_cookie=0 signature=n/a error-name=n/a error-message=n/a
Got message type=signal sender=org.freedesktop.DBus destination=:1.142 path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=2 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=signal sender=org.freedesktop.DBus destination=:1.142 path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=3 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=4 reply_cookie=2 signature=u error-name=n/a error-message=n/a
Successfully acquired requested service name.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=5 reply_cookie=3 signature=n/a error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.login1',path='/org/freedesktop/login1',interface='org.freedesktop.login1.Manager',member='PrepareForSleep' successfully installed.
Got message type=method_call sender=:1.143 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=ResolveHostname cookie=2 reply_cookie=0 signature=isit error-name=n/a error-message=n/a
Looking up RR for _gateway IN A.
Looking up RR for _gateway IN AAAA.
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=4 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetNameOwner cookie=5 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=7 reply_cookie=5 signature=s error-name=n/a error-message=n/a
Switching to DNS server 10.x.x.x for interface enp11s0.
Cache miss for _gateway.site.company.com IN A
Transaction 39567 for <_gateway.site.company.com IN A> scope dns on enp11s0/*.
Using feature level UDP+EDNS0 for transaction 39567.
Using DNS server 10.x.x.x for transaction 39567.
Sending query packet with id 39567.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=6 reply_cookie=4 signature=n/a error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.DBus',path='/org/freedesktop/DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0=':1.143' successfully installed.
Timeout reached on transaction 39567.
Retrying transaction 39567.
Switching to DNS server 10.x.x.x for interface enp11s0.
Cache miss for _gateway.site.company.com IN A
Transaction 39567 for <_gateway.site.company.com IN A> scope dns on enp11s0/*.
Using feature level UDP+EDNS0 for transaction 39567.
Using DNS server 10.x.x.x for transaction 39567.
Sending query packet with id 39567.
Processing incoming packet on transaction 39567 (rcode=NXDOMAIN).
Verified we get a response at feature level UDP+EDNS0 from DNS server 10.x.x.x.
Added NXDOMAIN cache entry for _gateway.site.company.com IN ANY 900s
Transaction 39567 for <_gateway.site.company.com IN A> on scope dns on enp11s0/* now complete with <rcode-failure> from network (unsigned).
Sent message type=method_return sender=n/a destination=:1.143 path=n/a interface=n/a member=n/a cookie=6 reply_cookie=2 signature=a(iiay)st error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=7 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Freeing transaction 39567.
Got message type=method_call sender=:1.144 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=ResolveHostname cookie=2 reply_cookie=0 signature=isit error-name=n/a error-message=n/a
Looking up RR for gateway IN A.
Looking up RR for gateway IN AAAA.
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=8 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetNameOwner cookie=9 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=9 reply_cookie=9 signature=s error-name=n/a error-message=n/a
Cache miss for gateway.site.company.com IN A
Transaction 23433 for <gateway.site.company.com IN A> scope dns on enp11s0/*.
Using feature level UDP+EDNS0 for transaction 23433.
Using DNS server 10.x.x.x for transaction 23433.
Sending query packet with id 23433.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=8 reply_cookie=8 signature=n/a error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.DBus',path='/org/freedesktop/DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0=':1.144' successfully installed.
Processing incoming packet on transaction 23433 (rcode=NXDOMAIN).
Added NXDOMAIN cache entry for gateway.site.company.com IN ANY 900s
Transaction 23433 for <gateway.site.company.com IN A> on scope dns on enp11s0/* now complete with <rcode-failure> from network (unsigned).
Sent message type=error sender=n/a destination=:1.144 path=n/a interface=n/a member=n/a cookie=10 reply_cookie=2 signature=s error-name=org.freedesktop.resolve1.DnsError.NXDOMAIN error-message='gateway' not found
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=11 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Freeing transaction 23433.
Got message type=method_call sender=:1.145 destination=org.freedesktop.resolve1 path=/org/freedesktop/resolve1 interface=org.freedesktop.resolve1.Manager member=ResolveHostname cookie=2 reply_cookie=0 signature=isit error-name=n/a error-message=n/a
Looking up RR for 0.clearlinux.pool.ntp.org IN A.
Looking up RR for 0.clearlinux.pool.ntp.org IN AAAA.
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=12 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetNameOwner cookie=13 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=11 reply_cookie=13 signature=s error-name=n/a error-message=n/a
Cache miss for 0.clearlinux.pool.ntp.org IN A
Transaction 16404 for <0.clearlinux.pool.ntp.org IN A> scope dns on enp11s0/*.
Using feature level UDP+EDNS0 for transaction 16404.
Using DNS server 10.x.x.x for transaction 16404.
Sending query packet with id 16404.
Got message type=method_return sender=org.freedesktop.DBus destination=:1.142 path=n/a interface=n/a member=n/a cookie=10 reply_cookie=12 signature=n/a error-name=n/a error-message=n/a
Match type='signal',sender='org.freedesktop.DBus',path='/org/freedesktop/DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0=':1.145' successfully installed.
Processing incoming packet on transaction 16404 (rcode=SUCCESS).
Added positive unauthenticated cache entry for 0.clearlinux.pool.ntp.org IN A 133s on enp11s0/INET/10.x.x.x
Added positive unauthenticated cache entry for 0.clearlinux.pool.ntp.org IN A 133s on enp11s0/INET/10.x.x.x
Added positive unauthenticated cache entry for 0.clearlinux.pool.ntp.org IN A 133s on enp11s0/INET/10.x.x.x
Added positive unauthenticated cache entry for 0.clearlinux.pool.ntp.org IN A 133s on enp11s0/INET/10.x.x.x
Transaction 16404 for <0.clearlinux.pool.ntp.org IN A> on scope dns on enp11s0/* now complete with <success> from network (unsigned).
Sent message type=method_return sender=n/a destination=:1.145 path=n/a interface=n/a member=n/a cookie=14 reply_cookie=2 signature=a(iiay)st error-name=n/a error-message=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=15 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Freeing transaction 16404.
^CBus bus-api-resolve: changing state RUNNING → CLOSED
Removing scope on link enp11s0, protocol dns, family *
Removing scope on link *, protocol dns, family *

```

**Comment by puneetse** [comment]
It looks like a known new regression upstream with repeat symptoms from an older bug . Some discussion and details here: https://github.com/systemd/systemd/issues/11391#issuecomment-488737464

**Comment by busykai** [comment]
I was able to reproduce the issue and pushed the fix. Should be gone with the next release out.

The users impacted are only those running `systemd-networkd`. Given that some run NetworkManager and not everybody has a domain configured, the exposure is limited.

Thank you @puneetse for the detailed info, it helped!

**Comment by miguelinux** [comment]
@busykai  your patch is not in series file :(, it will be lost in a new `autospec`

**Comment by busykai** [comment]
@miguelinux, 🤣 thanks! i did not even look around and my false memory told me it was not managed by `autospec`. TIL maybe don't fix stuff after 11pm?

**Comment by miguelinux** [comment]
> @miguelinux, 🤣 thanks! i did not even look around and my false memory told me it was not managed by `autospec`. TIL maybe don't fix stuff after 11pm?
> 

It is included in the morning build, lets see the results and just fix
it with autospec.

**Comment by busykai** [comment]
yeah, i know.

**Comment by puneetse** [comment]
30030 is working for me after restarting systemd-networkd

**Comment by busykai** [comment]
Closing.  Fixed by clearlinux-pkgs/systemd@e82c0fde3960 .
